app/controllers/items_controller.rb

```diff
@@ -1,82 +1,47 @@
 class ItemsController < ApplicationController
   # before_action :set_item, only: %i[ show edit update destroy ]
 
-  # GET /items or /items.json
   def index
     @merchant = Merchant.find(params[:merchant_id])
   end
 
-  # GET /items/1 or /items/1.json
   def show
     @merchant = Merchant.find(params[:merchant_id])
     @item = Item.find(params[:id])
   end
 
-  # GET /items/new
   def new
     @merchant = Merchant.find(params[:merchant_id])
   end
 
-  # GET /items/1/edit
   def edit
     @item = Item.find(params[:id])
     @merchant = Merchant.find(params[:merchant_id])
   end
 
-  # POST /items or /items.json
   def create
-    merchant = Merchant.find(params[:merchant_id])
     item = Item.new(item_params)
     item.save
-    redirect_to "/merchants/#{merchant.id}/items"
+    redirect_to merchant_items_path(params[:merchant_id])
   end
 
-  # PATCH/PUT /items/1 or /items/1.json
   def update
-    merchant = Merchant.find(params[:merchant_id])
     item = Item.find(params[:id])
-    if params[:name] == "" || params[:description] == "" || params[:unit_price] == ""
-      redirect_to "/merchants/#{merchant.id}/items/#{item.id}/edit"
-      flash[:alert] = "You cannot leave any field blank."
+    if params[:status] == 'Disabled'
```

Comment:
Nice refactor to make the #status_change more RESTful by adding to the #update action.